app/scripts/modules/azure/securityGroup/configure/CreateSecurityGroupCtrl.js

```diff
@@ -21,29 +36,147 @@ module.exports = angular.module('spinnaker.azure.securityGroup.create.controller
     };
 
+    function onApplicationRefresh() {
+      // If the user has already closed the modal, do not navigate to the new details view
+      if ($scope.$$destroyed) {
+        return;
+      }
+      $modalInstance.close();
+      var newStateParams = {
+        name: $scope.securityGroup.name,
+        accountId: $scope.securityGroup.credentials || $scope.securityGroup.accountName,
+        region: $scope.securityGroup.regions[0],
+        provider: 'azure',
+      };
+      if (!$state.includes('**.securityGroupDetails')) {
+        $state.go('.securityGroupDetails', newStateParams);
+      } else {
+        $state.go('^.securityGroupDetails', newStateParams);
+      }
+    }
+
+    function onTaskComplete() {
+      application.refreshImmediately();
+      application.registerOneTimeRefreshHandler(onApplicationRefresh);
+    }
+
+    $scope.taskMonitor = taskMonitorService.buildTaskMonitor({
+      application: application,
+      title: 'Creating your security group',
+      modalInstance: $modalInstance,
+      onTaskComplete: onTaskComplete,
+    });
+
+    $scope.securityGroup = securityGroup;
+
+    ctrl.accountUpdated = function() {
+      accountService.getRegionsForAccount($scope.securityGroup.credentials).then(function(regions) {
+        $scope.regions = regions;
+        $scope.securityGroup.regions = regions;
+        ctrl.updateName();
+      });
+    };
+
+    ctrl.cancel = function () {
+      $modalInstance.dismiss();
+    };
+
+    ctrl.updateName = function() {
+      var securityGroup = $scope.securityGroup,
+        name = application.name;
+      if (securityGroup.detail) {
+        name += '-' + securityGroup.detail;
+      }
+      securityGroup.name = name;
+      $scope.namePreview = name;
+    };
 
     ctrl.upsert = function () {
-      ctrl.mixinUpsert('Create');
+      $scope.taskMonitor.submit(
+        function() {
+          let params = {
+            cloudProvider: 'azure',
+            appName: application.name,
+            securityGroupName: $scope.securityGroup.name,
+            region: $scope.securityGroup.region,
+            subnet : 'none',
+            vpcId: 'null'
+            };
+          $scope.securityGroup.type = 'upsertSecurityGroup';
+
+          return azureSecurityGroupWriter.upsertSecurityGroup($scope.securityGroup, application, 'Create', params);
+        }
+      );
+    };
+
+    ctrl.addRule = function(ruleset) {
+      ruleset.push({
+        name: $scope.securityGroup.name + '-Rule' + ruleset.length,
+        priority: ruleset.length == 0 ? 100 : 100 * (ruleset.length + 1),
+        protocol: 'tcp',
+        access: 'Allow',
+        direction: 'InBound',
+        sourceAddressPrefix: '*',
+        sourcePortRange: '*',
+        destinationAddressPrefix: '*',
+        destinationPortRange: '7001-7001',
+        startPort: 7001,
+        endPort: 7001
+      });
+    };
+
+    ctrl.portUpdated = function(ruleset, index)
+    {
```

Comment:
Brace style is a little inconsistent - not breaking anything, just something that's easier to clean up and standardize on earlier than later.

I will probably add an eslint rule later to warn against non one-true-brace-style braces, but won't prevent code compilation.